dragonfly/implementations/roach_daq_run_interface.py

```diff
@@ -0,0 +1,338 @@
+'''
+'''
+
+from __future__ import absolute_import
+
+
+# standard imports
+import logging
+import time
+import os
+from .daq_run_interface import DAQProvider
+# internal imports
+from dripline import core
+
+__all__ = []
+
+logger = logging.getLogger(__name__)
+
+
+
+
+__all__.append('ROACH1ChAcquisitionInterface')
+class ROACH1ChAcquisitionInterface(DAQProvider):
+    '''
+    A DAQProvider for interacting with ROACH-Psyllid DAQ
+    '''
+    def __init__(self,
+                 psyllid_interface='psyllid_interface',
+                 daq_target = 'roach2_interface',
+                 channel = 'a',
+                 filename_prefix = 'psyllid',
+                 hf_lo_freq = None,
+                 **kwargs
+                ):
+
+        DAQProvider.__init__(self, **kwargs)
+
+        self.psyllid_interface = psyllid_interface
+        self.daq_target = daq_target
+        self.filename_prefix = filename_prefix
+        self.status_value = None
+        self.channel_id = channel
+        self.freq_dict = {self.channel_id: None}
+        self._max_duration = 0
+        self._run_time = 1
+        self._run_name = "test"
+        self.payload_channel = {'channel':self.channel_id}
+
+        if hf_lo_freq is None:
+            raise core.exceptions.DriplineValueError('the Psyllid acquisition interface requires a "hf_lo_freq" in its config file')
+        self._hf_lo_freq = hf_lo_freq
+
+
+
+    def _finish_configure(self):
+        logger.info('Doing setup checks...')
+        self._check_psyllid_instance()
+        
+        if self.status_value == 0:
+            result = self.provider.cmd(self.psyllid_interface, 'activate', payload = self.payload_channel)
+            self.status_value = self.provider.cmd(self.psyllid_interface, 'request_status', payload = self.payload_channel)['values'][0]
+                
+        if self._check_roach2_status()==True:
+            logger.info('Setting Psyllid central frequency identical to ROACH2 central frequency')
+            freqs = self._get_roach_central_freqs()
+            self.central_frequency = freqs[self.channel_id]
+
+
+    @property
+    def is_running(self):
+        self.status_value = self.provider.cmd(self.psyllid_interface, 'request_status', payload = self.payload_channel)
+        if self.status_value==5:
+            return True
+        else:
+            return False
+
+    @property
+    def max_duration(self):
+        return self._max_duration
+
+    @max_duration.setter
+    def max_duration(self, duration):
+        self._max_duration = duration
+
+
+    def _check_roach2_status(self):
```

Comment:
this method behavior is somewhat unintuitive. I'd expect "check_status" to either i) return the status, or ii) return either nothing or raise an exception if the status isn't good. It appears that the check is actually if the roach is running, could the method be `def _check_roach2_is_running(self)` instead?

I don't think I missed other cases of this, but the comment should apply elsewhere if I missed it.